### DISCUSSION TITLE
Way out of my depth...hope you don't mind me posting

### DISCUSSION THREAD

**Discussion by ShedSimoveGitHub** [katspaugh/wavesurfer.js#2672] [Q&A]
Hi Katspaugh (and everyone),
Firstly Katspaugh, thank you for making your brilliant Wavesurfer plugin.
I was thrilled to incorporate it into my new site, [earwonk.com](url).
And actually, the brand identity of earwonk is inspired by your waveforms. Fantastic.
I'm not a Coder, apologies for that, so I'm really out of my depth, and I'm having certain problems with lag and loading of the waveforms, and sometimes volume too, mostly on iPhone. 
I've been shown the error messages attached. Can you help please? 
And is there anyone on this forum who may be able to contact me (Shed Simove on most platforms, including Linkedin) about solving some of the issues the site is getting (paid)- and collaborating with me on my earwonk platform please.
All my best and have a superb day,
Shed Simove
![Screenshot 2023-02-22 at 16 42 21](https://user-images.githubusercontent.com/122627380/221000542-1112a3bf-073e-4ad0-b05f-cebafa545d8f.jpg)
![Screenshot 2023-02-22 at 16 42 32](https://user-images.githubusercontent.com/122627380/221000550-92e288f3-284d-4e1d-9a91-1c36b968c6e1.jpg)
![Screenshot 2023-02-22 at 16 44 18](https://user-images.githubusercontent.com/122627380/221000556-66b01116-0126-4e1c-a57d-3ca2c0520ab6.jpg)
![Screenshot 2023-02-22 at 16 41 56](https://user-images.githubusercontent.com/122627380/221000558-dbe0a6ba-ad88-4034-b246-553bf3887991.jpg)

**Comment by katspaugh** [comment]
Thanks Shed! Looks like you're also using the microphone plugin?
Please message me at katspaugh@gmail.com, we can organize something.

**Reply by ShedSimoveGitHub** [reply]
Thanks so much Katspaugh - incredibly helpful and appreciated indeed.

**Comment by entonbiba** [comment]
@ShedSimoveGitHub the screenshot error could mean the div element id is not present when wavesurfer tries to render it. Regarding the lag on iphone it would depend on how large the file is also if you already have the peaks data available for loading up front to avoid long loads of audio data rendering. Newer browsers and mobile devices require the user to initiate the audio by pressing a button.

The microphone plugin likely needs a rewrite to support the audioWorklet but I have a sample of how to record microphone audio and have the data as blob audio which you can then use with wavesurfer for rendering. Let us know if that helps, I can look into upgrading the microphone plugin soon.

https://www.entonbiba.com/codes/record-microphone-with-web-audio

**Reply by ShedSimoveGitHub** [reply]
Enton - thank you SO much for your fantastic reply. I'd love to explore your exciting proposition of upgrading the microphone plugin, thank you.
